app/context/extra_keyboard/index.tsx

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {createContext, useCallback, useContext, useEffect, useState} from 'react';
+import {Keyboard, Platform} from 'react-native';
+import Animated, {KeyboardState, useAnimatedKeyboard, useAnimatedStyle, useDerivedValue, useSharedValue, withTiming} from 'react-native-reanimated';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+
+import {Screens} from '@constants';
+import {useIsTablet} from '@hooks/device';
+import NavigationStore from '@store/navigation_store';
+import {preventDoubleTap} from '@utils/tap';
+
+import type {AvailableScreens} from '@typings/screens/navigation';
+
+export type ExtraKeyboardContextProps = {
+    isExtraKeyboardVisible: boolean;
+    component: React.ReactElement|null;
+    showExtraKeyboard: (component: React.ReactElement|null) => void;
+    hideExtraKeyboard: () => void;
+    registerTextInputFocus: () => void;
+    registerTextInputBlur: () => void;
+};
+
+// This is based on the size of the tab bar
+const KEYBOARD_OFFSET = -77;
+
+export const ExtraKeyboardContext = createContext<ExtraKeyboardContextProps|undefined>(undefined);
+
+const useOffetForCurrentScreen = (): number => {
```

Comment:
If I read this right, this starts at 0, and the moment it changes into tablet in the home screen, gets set to `-77`. But from that point onward, it never resets back to 0. Is that intended?

If it is not intended, and should return back to 0 when it is not tablet / it is not in the home screen, `offset` doesn't need to be a state, since it can be defined by the rest of the props without triggering a re-render.